js/memory.js

```diff
@@ -1,9 +1,35 @@
 class MemoryGame {
   constructor(cards){
     this.cards = cards;
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
     // add the rest of the class properties here
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
+
+  shuffleCards() {
+    
+    for (let i = this.cards.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [this.cards[i], this.cards[j]] = [this.cards[j], this.cards[i]];
+  }   
+}
+  
+  checkIfPair(card1, card2) {
+    if (card1 === card2) {
+      this.pairsGuessed++
+      return true;
+    } else {
+      this.pairsClicked++ 
+      return false;
+    }
+   }
+
+  isFinished() {
+    if (this.pairsGuessed === 8) {
```

Comment:
```suggestion
    if (this.pairsGuessed === this.cards.length / 2) {
```
Podrías haber comprobado de esta manera y ya te daba igual el número de cartas que Jasmine le pasa.